library/core/src/ptr/const_ptr.rs

```diff
@@ -14,12 +14,13 @@ impl<T: ?Sized> *const T {
     ///
     /// ## Behavior during const evaluation
     ///
-    /// When this function is used during const evaluation, it may return `false` for pointers
-    /// that turn out to be null at runtime. Specifically, when a pointer to some memory
-    /// is offset beyond its bounds in such a way that the resulting pointer is null,
-    /// the function will still return `false`. There is no way for CTFE to know
-    /// the absolute position of that memory, so we cannot tell if the pointer is
-    /// null or not.
+    /// If this method is used during const evaluation, and `self` is a pointer
+    /// that is offset beyond the bounds of the memory it initially pointed to,
+    /// then this method might or might not panic. Otherwise, it will not panic.
+    ///
+    /// This is because CTFE only knows `self`'s offset relative to the memory
+    /// it initially pointed to, and not its absolute address. This might or
+    /// might not be sufficient to determine nullness.
```

Comment:
The two paragraphs are still both talking about the same thing, which is odd. There's a single point that we are trying to make here, it should only need one paragraph. And then maybe we have a second paragraph about a different point. So here's my proposal:
```suggestion
    /// then there might not be enough information to determine whether the
    /// pointer is null. This is because the absolute address in memory is not
    /// known at compile time. If the nullness of the pointer cannot be determined,
    /// this method will panic.
    ///
    /// In-bounds pointers are never null, so the method will never panic
    /// for such pointers.
```